### ISSUE TITLE
Allow custom theme/colors; dark theme not quite right

### ISSUE THREAD

**Issue by gnrlus** [keepassxreboot/keepassxc#7493]
## Summary
I have KeepassXC 2.4.3 installed which adopts my Qt5 custom color scheme and it's great. But when I run 2.6.6 I'm limited to 4 theme presets, none of which I like. Could you add an option to allow custom colors as set by Qt5 configuration tool?

## Examples
See attached screenshots.

## Context
The eyes are sensitive to blue and should be avoided. Since white and light gray contain a lot of blue they should be avoided. Blue disrupts the brain's sleep functions and has even created a market for blue filtering glasses (that make everything appear yellow). I have found that yellow (almost pure, but not 100% green, and not 100% bright) for text provides the perfect balance of stimulation and ease on the eyes. It stands out and is easy for the eye to find but does not blast the eye with radiation either, and it's just a lovely warm color. Cold colors tend to have a negative psychological effect and the I don't like the green buttons. The background should be as dim as possible. People who sit at the computer all day are getting too much radiation on their retinas, even from mid gray backgrounds and a lot of people with eye problems cannot tolerate this particularly.

![this_is_better](https://user-images.githubusercontent.com/90088274/156714809-f40cf8b6-fd2e-4904-8f11-857e5f00cdcc.png)
![drag,_more_eye_strain](https://user-images.githubusercontent.com/90088274/156714831-ef9020f9-9543-46f0-a4f4-94e025e8774f.png)

**Comment by phoerious** [comment]
If you want more yellow, reduce the colour temperature of your screen. This is not the app's task to do and I doubt that KeePassXC is the app you are looking at all day. The contrast ratio of your yellow colour is also lower than what we provide.
Also, if you are worried about your sleep schedule, reduce your screen time. Turning everything more yellow has very little effect (there is some, but it's very minor).

**Comment by gnrlus** [comment]
What shouldn't be the app's task is controlling what colors the user gets stuck with. Why limit the user? Every other app I can conform to my own theme. Adjusting the color warmth of the monitor throws everything off and is not a solution. There is more contrast between yellow and gray than gray and gray and green and gray. For me the yellow has a profound effect. It probably varies person to person because everyone has different brains with different bandwidth allocated to visual signal processing. I cannot reduce my screen time; I have work to do. I find eliminating blue from my screen eliminates sleep interference sufficiently. Thank you.

**Comment by phoerious** [comment]
No, the contrast ratio between the original background and text colours is between 6.9:1 and 8.3:1. Yours is at about 5.3:1, so 2-3 full points less.
And no, not every app lets you control its font colours, in fact hardly any app lets you do that. If you want to control the UI appearance, use the Classic theme, which will adopt your system UI colour scheme.

**Comment by gnrlus** [comment]
There is a difference between contrast in terms of luminosity and hue. Clearly, yellow stands out more against mid-gray than light-gray on mid-gray. The eye is sensitive to color, not just luminosity. And actually, if I'm going to use this app I will be staring it at all day because I enter passwords all day long. Virtually every app I have lets me control the colors, though you are correct a lot of apps don't let you control--they should; that's why distros have theme customization built in. I selected 'Classic' and it just changes everything to white background and black text. It does not comply to my custom theme. If you can help debug/fix that I'd appreciate it. Should I open a new ticket?

**Comment by phoerious** [comment]
The contrast ratio is in compliance with WCAG AA and AAA standards. We will not change those.
If Classic displays a light theme, then your theme is not set up correctly. If your desktop is GTK-based, use Qt5ct for setting the Qt theme.

**Comment by droidmonkey** [comment]
If you are using the snap to get 2.6.6 then that is the cause of themeing issues in classic mode. Use the appimage instead.

**Comment by gnrlus** [comment]
Well I don't care what the standards say, they're harder to look at, and those are probably just minimum standards, not optimal. None of my other Qt5 apps have this issue. My Qt5 theme works everywhere else. It was working perfectly in 2.4.3. I am using the .AppImage for 2.6.6. Maybe there is something about how the AppImage loads that doesn't honor the Qt5 theme? My 2.4.3 is from Linux Mint repo.

**Comment by phoerious** [comment]
The AppImage uses Qt's default Fusion style as its classic theme. Install 2.6.6 from our PPA.

**Comment by gnrlus** [comment]
Linux Mint forum advises against PPAs / 3rd party repos and for a good reason. Nearly everyone I have added runs over my system files with different versions that cause absolute mayhem. (The last one I tried was a super popular one, MySQL Workbench, and it totally hosed my system, causing my calculator to take 30 seconds to load!) The forum scolded me good for even thinking about using 3p repos. I said, "Okay, okay" and now I use AppImages as much as possible. I like AppImages because they are capable of including all dependencies and AFAIK don't attempt to replace the user's default system files. The Qt 'style' is just that--the style of graphics, but the colors are a different setting called 'Palette'. My Qt5 lets me choose between 'Default' and 'Custom', and I've created and selected my own 'custom' scheme. When I switch 2.6.6 to 'Classic' I noticed the style changes. It is like 'Automatic' setting but with a different style--and no green buttons...
![Screenshot_2022-03-04_14-01-34](https://user-images.githubusercontent.com/90088274/156833428-9dc35129-5390-423d-8d59-c7c475c753df.png)
